subworkflows/align_short/subworkflows/local/markdup_stats.nf

```diff
@@ -0,0 +1,49 @@
+//
+// Merge and Markdup all alignments at specimen level
+// Convert to CRAM and calculate statistics
+//
+include { SAMTOOLS_SORT } from '../../modules/nf-core/modules/samtools/sort/main'
+include { MARKDUPLICATE } from '../../subworkflows/local/markduplicate'
+include { CONVERT_STATS } from '../../subworkflows/local/convert_stats'
+
+workflow MARKDUP_STATS {
+    take:
+    aln // channel: [ val(meta), [ sam ] ]
+    fasta // channel: /path/to/fasta
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Sort SAM and convert to BAM
+    SAMTOOLS_SORT ( aln )
+    ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())
+
+    // Collect all BWAMEM2 output by sample name
+    SAMTOOLS_SORT.out.bam
+    .map { meta, bam ->
+    new_meta = meta.clone()
```

Comment:
Yes if I change the `meta.id` without the clone, it messes up the `SAMTOOLS_SORT.out.bam` queue channel value which is of indeterminate length, since we don't know how many samples in advance. I kept getting an error, so Mahesh gave me this suggestion. Can't find the thread anymore unfortunately. But I removed the `clone` bit and ran to just see the error again.

I get errors such as `IndexOutOfBoundsException: fromIndex = -1` or an empty `meta`. Weird errors.